### ISSUE TITLE
Can't create more than 9 pipes

### ISSUE THREAD

**Issue by ramaza** [luigirizzo/netmap#519]
Netmap (v11.4) only allows to open 9 pipes with default settings. I wasn't able to increase that limit by playing around with the Sysctl variables. It feels like the Sysctl variables only have an effect on hardware interfaces but not Netmap pipes.

What do I need to do (maybe in the code) to allow more than 100 Netmap pipes?

By the way, how much memory does a single pipe require. My calculation says it should be around 8MB. Is this correct?

**Comment by giuseppelettieri** [comment]
Is there any message in the kernel logs that mentions which allocation is failing?

**Comment by giuseppelettieri** [comment]
The interesting message is 'no more 2_ring objects', which means that you have exhausted the pre-allocated  ```nemap_ring``` objects in the port memory region (id 2, in this case). Any id greater than 1 denotes a private memory region. You can control the pre-allocations for private memory regions using the ```priv_*_num``` parameters. You need 8 ```nemap_ring```s for each pipe, 1 TX and 1 RX for each endpoint plus two dummy host rings per end point, for legacy reasons. You don't need buffers for the dummy host rings, however.

**Comment by ramaza** [comment]
I think I understand. As we later on need to add hardware interfaces to our setup it might be better to force the pipes into the global memory region with `...@1` syntax and increase `ring_num` instead of `priv_ring_num`. Right?

**Comment by giuseppelettieri** [comment]
Yes, you are right. I guess that using the global memory region is simpler in the current implementation.
